src/parser.ts

```diff
@@ -900,17 +902,21 @@ export class Parser extends DiagnosticEmitter {
 
     var typeParameters = new Array<TypeParameterNode>();
     if (!tn.skip(Token.GREATERTHAN)) {
-      do {
+      while (!tn.skip(Token.GREATERTHAN)) {
```

Comment:
Maybe you have an idea how the initial double-check for GREATERTHAN could be avoided? :)

Reply:
I guess do/while handles that pretty nicely here. Will update.

Reply:
Actually, I'll just do a length check for the "no type parameters" case so this looks like the other cases.

Reply:
✅

Reply:
Thanks, looking good :)